src/robot_state_publisher.cpp

```diff
@@ -148,9 +148,7 @@ RobotStatePublisher::RobotStatePublisher(const rclcpp::NodeOptions & options)
       &RobotStatePublisher::callbackJointState, this,
       std::placeholders::_1));
 
-  // trigger to publish fixed joints
-  timer_ = this->create_wall_timer(
-    publish_interval_ms_, std::bind(&RobotStatePublisher::publishFixedTransforms, this));
```

Comment:
IIUC if `use_tf_static` is False then the fixed transforms must be published regularly or they will not be received by late joining subscribers.

I think creating the timer only when `use_tf_static` is False would be a backportable change.

Reply:
Fixed at https://github.com/ros/robot_state_publisher/pull/175/commits/091935468e0792b79b846d6fe401a2ef19275b56